README.md

```diff
@@ -155,6 +156,98 @@ archival storage efforts – but the hash URI identifier allows us to
+#> 3  hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 4  hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 5  hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 6  hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 7  hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 8  hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 9  hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 10 hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 11 hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 12 hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 13 hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#> 14 hash://sha256/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#>                                                                                    source
+#> 1  https://github.com/espm-157/climate-template/releases/download/data/vostok.icecore.co2
+#> 2                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 3                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 4                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 5                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 6                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 7                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 8                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 9                                 http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 10                       https://cdiac.ess-dive.lbl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 11                                http://cdiac.ornl.gov/ftp/trends/co2/vostok.icecore.co2
+#> 12 https://github.com/espm-157/climate-template/releases/download/data/vostok.icecore.co2
+#> 13 https://github.com/espm-157/climate-template/releases/download/data/vostok.icecore.co2
+#> 14 /tmp/RtmpckfTLc/94/12/9412325831dab22aeebdd674b6eb53ba6b7bdd04bb99a4dbb21ddff646287e37
+#>                   date
+#> 1  2020-02-21 17:33:41
+#> 2  2020-02-20 20:31:33
+#> 3  2020-02-19 16:37:55
+#> 4  2020-02-17 19:36:18
+#> 5  2020-02-16 04:59:47
+#> 6  2020-02-15 01:18:15
+#> 7  2020-02-13 23:43:35
+#> 8  2020-02-12 22:54:39
+#> 9  2020-02-11 20:52:53
+#> 10 2020-02-08 05:05:34
+#> 11 2020-02-08 04:45:00
+#> 12 2020-02-21 17:36:01
+#> 13 2020-02-21 17:36:02
+#> 14 2020-02-21 17:36:02
+```
+
+The `retreive()` verb (subject to change), provides a wrapper around
+`query` that always returns exactly one local path to requested content.
```

Comment:
why only one local path? What if you have multiple copies in various local stores?

Reply:
Maybe this isn't the role of `retrieve()`, but a core use case for me is being able to do stuff like:

```r
read_csv(retrieve("hash://sha256/xxx"))
```

Obviously for that to work, the function has to have some logic on how to resolve multiple copies!  Not sure I nailed the logic yet, but so far it prefers local copies (because downloads are slow), and falls back on remote copies (which it also verifies are the content requested, or else it tries the next registered location until it gives up).  

Note that I think this case speaks to a reasonable audience of R users who like DOIs and have always wanted to do:

```r
read_csv("doi:xxx")
```

which of course never works out of the box because dois usually only take you to landing pages.

Reply:
I was imagining ```pin``` doing this logic.

As far as read_csv("doi:xxx") goes. . . in practice data doi link to one or more versions of data (landing page html, some referenced data files, etc). So, read_csv("doi:xxx") really means read the doi only if this doi is associated with a single data file. You'd need someone (like preston or other crawlers, inferrers), to interpret the doi and make explicit links to data. That interpretation can be associated to the doi and then that association can be registered. Then, you can do things like:

```
list_of_data_references <- query("doi:xxx", accordingTo = "preston") 
# pick the first one
readLines(retrieve(list_of_data_references[0].url))
```

Reply:
By this logic, I mean:

```
read_csv(pin("hash://sha256/xxx"))
```